pkg/datapath/loader/loader.go

```diff
@@ -119,6 +119,20 @@ func (l *Loader) reloadDatapath(ctx context.Context, ep datapath.Endpoint, dirs
 			}
 			return err
 		}
+		if ep.RequireEgressProg() {
+			if err := graftDatapath(ctx, ep.MapPath(), objPath, symbolToEndpoint, 1); err != nil {
+				scopedLog := ep.Logger(Subsystem).WithFields(logrus.Fields{
+					logfields.Path: objPath,
+				})
+				// Don't log an error here if the context was canceled or timed out;
+				// this log message should only represent failures with respect to
+				// loading the program.
+				if ctx.Err() == nil {
+					scopedLog.WithError(err).Warn("JoinEP: Failed to load program")
+				}
+				return err
+			}
+		}
```

Comment:
It seems this code is duplicated from the previous lines. Can't you do it like:
```golang
key := "0"
if ep.RequireEgressProg() {
   key = "1"
}
if err := graftDatapath(ctx, ep.MapPath(), objPath, symbolFromEndpoint, key); err != nil {
....
```

Reply:
Thanks for you advice, I will change it.

Reply:
Hi @aanm , I read this code again and found that it cannot be modified as you suggested. 

Load bpf for key "0" is always required. But load bpf for  key "1" is only required when ```ep.RequireEgressProg()``` return true.  As follow:

```
if err := graftDatapath(..., 0); err != nil {
   ...
}
if ep.RequireEgressProg() {
    if err := graftDatapath(..., 1); err != nil {
       ...
    }
}
```

They are different in logic.

Reply:
Ah you are right, sorry about it!